src/pybacked.rs

```diff
@@ -85,6 +86,20 @@ impl FromPyObject<'_> for PyBackedStr {
     }
 }
 
+#[cfg(any(Py_3_10, not(Py_LIMITED_API)))]
+impl ToPyObject for PyBackedStr {
+    fn to_object(&self, py: Python<'_>) -> Py<PyAny> {
+        self.storage.clone_ref(py)
+    }
+}
+
+#[cfg(any(Py_3_10, not(Py_LIMITED_API)))]
```

Comment:
The cfgs are unfortunate; I think it should still be possible to write the implementations even if it's a bit more work.

You are right to call out the complexity though; we should add some tests for these methods :)